opus/application/static_media/js/mutationObserver.js

```diff
@@ -140,18 +139,35 @@ var o_mutationObserver = {
 
         // ps in select metadata modal
         let metadataSelectorObserver = new MutationObserver(function(mutationsList) {
-            mutationsList.forEach((mutation, idx) => {
-                adjustMetadataSelectorMenuPS();
-                adjustSelectedMetadataPS();
-            });
+            adjustMetadataSelectorMenuPS();
+            adjustSelectedMetadataPS();
         });
 
         let metadataSelectorMenuObserver = new MutationObserver(function(mutationsList) {
             let lastMutationIdx = mutationsList.length - 1;
             mutationsList.forEach((mutation, idx) => {
                 if (idx === lastMutationIdx) {
                     if (mutation.target.classList.value.match(/submenu.collapse/)) {
-                        adjustMetadataSelectorMenuPS();
+                        // If a collapse/expand happened (attribute changes),
+                        // we only update ps at the last mutation when the animation
+                        // finishes and class/style are finalized.
+                        // Note we call the original version here, not the debounced
+                        // version, because we need the PS to be visible instantly in
+                        // order for scrollTop to work below.
+                        o_browse.adjustMetadataSelectorMenuPS();
+                        // if the collapse opens below the viewable area, move the scrollbar
+                        // only move the scrollbar if we open the menu item
+                        let lastElement = $(mutation.target).children().last();
```

Comment:
line 186 et all are exactly the same - function?

Reply:
Actually there are quite a few differences - one is doing things to the left side pane (with collapsing categories) and the other is doing things to the right pane (list of selected metadata). Although the code looks similar from a distance, almost every line is different. I looked to see if it would be possible to refactor, but I think it would just make the code more complicated.